yjit_iface.c

```diff
@@ -500,6 +534,67 @@ struct yjit_block_itr {
     VALUE list;
 };
 
+/* For each entry block associated with `iseq`, call the function pointer
+ * `cb`, passing the entry block along with the ctx pointer */
+static void
+each_entry_block_for_iseq(const rb_iseq_t *iseq, block_iter_t cb, void *ctx)
+{
+    rb_darray_for(iseq->body->yjit_blocks, version_array_idx) {
+        rb_yjit_block_array_t versions = rb_darray_get(iseq->body->yjit_blocks, version_array_idx);
+
+        rb_darray_for(versions, block_idx) {
+            block_t *block = rb_darray_get(versions, block_idx);
+
+            if (block->blockid.idx == 0) {
+                cb(block, ctx);
+            }
+        }
+    }
+}
```

Comment:
AFAIK the array is indexed by position in the iseq, so you wouldn't need to iterate through every possible block_idx.

Reply:
Ah ya, 🤦🏻‍♂️.  I'll fix it to just use the index and that should clean this function up.  Thank you!